examples/auctions/simple_open_auction.v.py

```diff
@@ -0,0 +1,76 @@
+# Open Auction
+
+# Auction params
+# Beneficiary recieves money from the highest bidder
+beneficiary: public(address)
+auction_start: public(timestamp)
+bidding_time: public(num(sec))
+
+# Current state of auction
+highest_bidder: public(address)
+highest_bid: public(num(wei))
+
+# Allowed withdrawals of previous bids
+pending_returns:(num(wei)[address])
+
+# Set to true at the end, disallows any change
+ended: public(bool)
+
+# Create a simple auction with `_bidding_time`
+# seconds bidding time on behalf of the
+# beneficiary address `_beneficiary`.
+def __init__(_beneficiary: address, _bidding_time: num(sec)):
+    self.beneficiary = _beneficiary
+    self.bidding_time = _bidding_time
+    self.auction_start = block.timestamp
+
+# Bid on the auction with the value sent
+# together with this transaction.
+# The value will only be refunded if the
+# auction is not won.
+@payable
+def bid():
```

Comment:
That's clever, that way you don't have to worry about refunding at the end. Although, this may end up costing more fees overall because more transactions occur per person? I'm picturing a bidding war where two people are going back and forth on an item, and the constant refund -> re-bid transaction fees might be more than the fees to add more value to your bid each time. Maybe there is a compromise where you maintain a 5 person queue, and you get refunded if you fall off of that queue, but if you're in the queue, the value adds? Lots of potential issues with that thought maybe. A very rich set of trades to explore there I think.